### PR TITLE
Tyr: add validaton of email on user creation

### PR DESCRIPTION
**Pull request by kinnou02**
it use the python package validate_email who check the format and
optionaly check the MX record of the DNS and can connect to the SMTP
server for check the user existance.

Two parameters are added in settings for managing these capabilities.

### CODE REVIEW

File: source/tyr/tyr/resources.py [hunk 1]
```diff
@@ -78,6 +79,12 @@ def post(self):
         parser.add_argument('email', type=unicode, required=True,
                 case_sensitive=False, help='email is required')
         args = parser.parse_args()
+
+        if not validate_email(args['email'],
+                          check_mx=current_app.config['EMAIL_CHECK_MX'],
+                          verify=current_app.config['EMAIL_CHECK_SMTP']):
+            return ({'error': 'email invalid'}, 400)
```
Comment (l-vincent-l) [inline source/tyr/tyr/resources.py:86]: Tu peux utiliser ça : http://flask.pocoo.org/docs/api/#flask.abort

File: source/tyr/tyr/resources.py [hunk 2]
```diff
@@ -95,6 +102,12 @@ def put(self, user_id):
...
```
Comment (l-vincent-l) [inline source/tyr/tyr/resources.py:109]: Pareil : http://flask.pocoo.org/docs/api/#flask.abort

File: source/tyr/tyr/resources.py (hunk 1 above)
Comment (l-vincent-l) [inline source/tyr/tyr/resources.py:86]: en fait : http://flask-restful.readthedocs.org/en/latest/api.html?highlight=abort#flask.ext.restful.abort

Comment (kinnou02) [inline source/tyr/tyr/resources.py:86]: et c'est quoi l'avantage?

**Pull request merged**